pkgs/development/haskell-modules/configuration-ghc-9.4.x.nix

```diff
@@ -214,4 +214,20 @@ in {
   # failing during the Setup.hs phase: https://github.com/gtk2hs/gtk2hs/issues/323.
   gtk2hs-buildtools = appendPatch ./patches/gtk2hs-buildtools-fix-ghc-9.4.x.patch super.gtk2hs-buildtools;
 
+  haskell-gi-base = __CabalEagerPkgConfigWorkaround super.haskell-gi-base;
+  haskell-gi = __CabalEagerPkgConfigWorkaround super.haskell-gi;
+  gi-adwaita = __CabalEagerPkgConfigWorkaround super.gi-adwaita;
+  gi-atk = __CabalEagerPkgConfigWorkaround super.gi-atk;
+  gi-cairo = __CabalEagerPkgConfigWorkaround super.gi-cairo;
+  gi-gdk = __CabalEagerPkgConfigWorkaround super.gi-gdk;
+  gi-gdkpixbuf = __CabalEagerPkgConfigWorkaround super.gi-gdkpixbuf;
+  gi-gdk_4_0_5 = __CabalEagerPkgConfigWorkaround super.gi-gdk_4_0_5;
+  gi-gio = __CabalEagerPkgConfigWorkaround super.gi-gio;
+  gi-glib = __CabalEagerPkgConfigWorkaround super.gi-glib;
+  gi-gmodule = __CabalEagerPkgConfigWorkaround (overrideCabal (drv: {libraryPkgconfigDepends = [pkgs.glib];}) super.gi-gmodule);
```

Comment:
What's going on here?

Reply:
gi-gmodule has the following code 
```nix
gi-gmodule" = callPackage
    ({ mkDerivation, base, bytestring, Cabal, containers, gi-glib
     , gmodule, haskell-gi, haskell-gi-base, haskell-gi-overloading
     , text, transformers
     }:
     mkDerivation {
       pname = "gi-gmodule";
       version = "2.0.3";
       sha256 = "043n3nyxy29chzc7xzhinp40yxazlikqcjdbm3pvh344jv7m5xjx";
       setupHaskellDepends = [ base Cabal gi-glib haskell-gi ];
       libraryHaskellDepends = [
         base bytestring containers gi-glib haskell-gi haskell-gi-base
         haskell-gi-overloading text transformers
       ];
       libraryPkgconfigDepends = [ gmodule ];
       description = "GModule bindings";
       license = lib.licenses.lgpl21Only;
     }) {gmodule = null;};
``` 
because gmodule is null it doesn't find its dependency glib which is necessary for compiling. on second thought it should probably be commented

Reply:
That is easily fixed properly in cabal2nix.

Reply:
Done in bb62d57a851, you should be able to rebase and simplify this rebase.